test/test.js

```diff
@@ -256,4 +256,154 @@ describe('ConfigGenerator', function () {
             });
         });
     });
+
+    it('should rewrite "define" calls if "namespace" option is present', function (done) {
+        var configGenerator = new ConfigGenerator({
+            args: [path.resolve(__dirname, 'modal')],
+            config: '',
+            filePattern: '**/namespace-define-skip*.js',
+            format: ['/_/g', '-'],
+            ignorePath: false,
+            moduleConfig: path.resolve(__dirname, 'modal/bower.json'),
+            moduleRoot: path.resolve(__dirname, 'modal'),
+            skipFileOverride: true,
+            namespace: 'Liferay'
+        });
+
+        configGenerator.process().then(function(config) {
+
+            assert.strictEqual(normalizeCR(config), normalizeCR(fs.readFileSync(path.resolve(__dirname,
+                'expected/expected-namespace-define-skip-override-config.js'), 'utf-8')));
+
+            var actual = fs.readFileSync(path.resolve(__dirname, 'modal/js/namespace-define-skip-override.es.js'), 'utf-8');
+            var expected = fs.readFileSync(path.resolve(__dirname, 'expected/expected-namespace-define-skip-override.es.js'), 'utf-8');
+
+            assert.strictEqual(normalizeCR(actual), normalizeCR(expected));
+
+            done();
+        });
+    });
+
+    it('should rewrite "define" calls and save the file if "namespace" option is present and "skipFileOverride" option is "false"', function (done) {
+        var configGenerator = new ConfigGenerator({
+            args: [path.resolve(__dirname, 'modal')],
+            config: '',
+            filePattern: '**/namespace-define-override*.js',
+            format: ['/_/g', '-'],
+            ignorePath: false,
+            moduleConfig: path.resolve(__dirname, 'modal/bower.json'),
+            moduleRoot: path.resolve(__dirname, 'modal'),
+            skipFileOverride: false,
+            namespace: 'Liferay'
+        });
+
+        configGenerator.process().then(function(config) {
+
+            assert.strictEqual(normalizeCR(config), normalizeCR(fs.readFileSync(path.resolve(__dirname,
+                'expected/expected-namespace-define-override-config.js'), 'utf-8')));
+
+            var actual = fs.readFileSync(path.resolve(__dirname, 'modal/js/namespace-define-override.es.js'), 'utf-8');
+            var expected = fs.readFileSync(path.resolve(__dirname, 'expected/expected-namespace-define-override.es.js'), 'utf-8');
+
+            assert.strictEqual(normalizeCR(actual), normalizeCR(expected));
+
+            done();
+        });
+    });
+
+    it('should rewrite "define" calls if "namespace" option is present and change the "namespace" if different', function (done) {
```

Comment:
We probably want to assert the exact opposite... that we are not rewriting anything that is not a pure `define` call